gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/accesspoint/manager/DefaultAccessPointManager.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.handlers.accesspoint.manager;
+
+import io.gravitee.common.event.EventManager;
+import io.gravitee.gateway.reactor.accesspoint.AccessPointEvent;
+import io.gravitee.gateway.reactor.accesspoint.ReactableAccessPoint;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicBoolean;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class DefaultAccessPointManager implements AccessPointManager {
+
+    private final EventManager eventManager;
+    private final Map<String, List<ReactableAccessPoint>> accessPoints = new ConcurrentHashMap<>();
+
+    @Override
+    public void register(ReactableAccessPoint reactableAccessPoint) {
+        AtomicBoolean added = new AtomicBoolean(false);
+        accessPoints.compute(
+            reactableAccessPoint.getEnvironmentId(),
+            (k, v) -> {
+                if (v == null) {
+                    v = new CopyOnWriteArrayList<>();
+                }
+                if (!v.contains(reactableAccessPoint)) {
+                    added.set(v.add(reactableAccessPoint));
```

Comment:
why don't you publish the event here instead of relying on an atomic boolean `if v.add(...) => deploy`